### PR TITLE
feat: create LuminatorError class

### PR DESCRIPTION
**Pull request by sundowndev**
Create a custom error class to handle Luminati related errors.

### CODE REVIEW

File: src/luminator.ts [hunk 1]
```diff
@@ -17,6 +17,8 @@ interface ILuminatorConfig {
 
 type LuminatorResponse<T = any> = AxiosResponse<T>;
 
+class LuminatorError extends Error {}
```
Comment (rimiti) [inline src/luminator.ts]: 👍

File: src/luminator.ts [hunk 2]
```diff
@@ -104,9 +106,7 @@ class Luminator {
    * @throws {Error}
```
Comment (rimiti) [inline src/luminator.ts]:
```
* @throws {LuminatorError}
```

File: __tests__/unit/index.ts [hunk 3]
```diff
@@ -77,17 +78,21 @@ describe('Luminator', () => {
     });
 
     it('Should fail with 404 status, with error message MAX_FAILURES_REQ', async () => {
-      const err = failWith(404, `failed status 404`);
-      mockAxios.mockRejectedValue(err);
+      const responseError =failWith(404, `failed status 404`)
```
Comment (rimiti) [inline __tests__/unit/index.ts]:
```
const responseError = failWith(404, `failed status 404`);
```

File: __tests__/unit/index.ts [hunk 4]
```diff
@@ -68,7 +69,7 @@ describe('Luminator', () => {
   let agent: Luminator = new Luminator('USERNAME', 'password');
 
   beforeEach(() => {
-    jest.restoreAllMocks();
+    jest.resetAllMocks();
```
Comment (rimiti) [inline __tests__/unit/index.ts]: 👍

File: __tests__/unit/index.ts [hunk 5]
```diff
@@ -101,16 +106,20 @@ describe('Luminator', () => {
...
             url: 'https://lumtest.com/myip.json',
           });
         } catch (e) {
-          expect(spy).toHaveBeenCalledTimes(7);
-          expect(e).toEqual(new Error('MAX_FAILURES_REQ threshold reached'));
+          err = e;
         }
 
+        expect(spy).toHaveBeenCalledTimes(7);
+        expect(err).toEqual(new Error('MAX_FAILURES_REQ threshold reached'));
```
Comment (rimiti) [inline __tests__/unit/index.ts]: Shouldn't be:

```
expect(err).toEqual(new LuminatorError('MAX_FAILURES_REQ threshold reached'));
```

File: __tests__/unit/index.ts [hunk 6]
```diff
@@ -134,14 +143,19 @@ describe('Luminator', () => {
...
         await agent.fetch({
           method: 'GET',
           url: 'https://lumtest.com/myip.json',
         });
       } catch (e) {
-        expect(e).toEqual(new Error('NON_AXIOS_ERROR'));
+        err = e;
       }
+
+      expect(err).toEqual(new Error('NON_AXIOS_ERROR'));
```
Comment (rimiti) [inline __tests__/unit/index.ts]: Same question

File: src/luminator.ts [hunk 7]
```diff
@@ -104,14 +106,12 @@ class Luminator {
...
    * @return {Promise<LuminatorResponse<unknown>>}
```
Comment (rimiti) [inline src/luminator.ts]:
```
* @return {Promise<LuminatorResponse>}
```

File: src/luminator.ts [hunk 8]
```diff
@@ -82,14 +84,14 @@ class Luminator {
...
-   * @throws {Error} non axios error
+   * @throws {LuminatorError} non axios error
    * @return {number}
    */
   private static getStatusFromAxiosError(error: AxiosError): number {
     try {
       return error.response.status;
     } catch (e) {
-      throw error;
+      throw new LuminatorError(`NON_AXIOS_ERROR`);
```
Comment (rimiti) [inline src/luminator.ts]: Not sure that really useful to use uppercase message.

File: src/luminator.ts [hunk 9]
```diff
@@ -104,14 +106,12 @@ class Luminator {
    * @throws {Error}
```
Comment (rimiti) [inline src/luminator.ts]:
```
 * @throws {LuminatorError}
```

File: src/luminator.ts (hunk 7 above)
Comment (rimiti) [inline src/luminator.ts]: Promise<LuminatorResponse>

File: src/luminator.ts (hunk 8 above)
Comment (rimiti) [reply src/luminator.ts]: its maybe better to :

```
throw new LuminatorError(e);
```

Comment (sundowndev) [reply src/luminator.ts]: The original error about the line above "cannot use property status of undefined". I don't think this a relevant error, is it ?

Comment (sundowndev) [reply src/luminator.ts]: It means the request has no status, just a proxy error

Comment (rimiti) [reply src/luminator.ts]: If it works for our needed, it's okay for me.

**Pull request merged**